com.unity.render-pipelines.core/Runtime/Textures/RTHandleSystem.cs

```diff
@@ -219,8 +219,10 @@ public void SetReferenceSize(int width, int height, MSAASamples msaaSamples, boo
 
             if (DynamicResolutionHandler.instance.HardwareDynamicResIsEnabled() && m_HardwareDynamicResRequested)
             {
-                float xScale = (float)DynamicResolutionHandler.instance.finalViewport.x / GetMaxWidth();
-                float yScale = (float)DynamicResolutionHandler.instance.finalViewport.y / GetMaxHeight();
+                var scaledFinalViewport = DynamicResolutionHandler.instance.GetScaledSize(DynamicResolutionHandler.instance.finalViewport);
+                var scaledMaxViewport = DynamicResolutionHandler.instance.GetScaledSize(new Vector2Int(GetMaxWidth(), GetMaxHeight()));
```

Comment:
This is a bit dangerous as it will cache the original size as GetMaxWidth which is not necessarily what we want. 

Is this to have the ceiling applied before the scale calculation? Because if not outside for the side effect of caching the original scale this is equivalent to 

finalViewport * fraction / maxViewport * fraction == finalViewport / maxViewport which as there before

Reply:
If it is indeed what you wanted to do (minus the caching), we might want to add a new API to get the same data without side effect.

Reply:
It is indeed what we want. The reason why that equality you typed is not the same is because of rounding:

round(finalViewport * fraction) / round(maxViewport * fraction) != finalViewport/maxViewport
The rations need to consider the partial pixel of the downsampled resolution, otherwise you get some weird lines in the edges.
Hope that made sense, I am now writting proper pure functions for those APIs. The side effect did cause some issues, thank you for pointing that out!

Reply:
Yup wrote in the comment that if it was rounding then yeah then we want (The equation was an _if not_ ) :D 

But yeah, we probably want something without side effect, shame we'd need to make that public. So I'd suggest to have an helper to just get the fraction and then something internal to HDRP only to do the same scaling as the handler.

Reply:
Added some new APIS. LMK what you think.